### DISCUSSION TITLE
Dart Language Server running analysis on every insert

### DISCUSSION THREAD

**Discussion by khoa-l** [LazyVim/LazyVim#1359] [Q&A]
I am trying to setup Flutter / Dart and am running into an issue with the Dart Language Server. On every insert action the Dart Language Server runs analysis. Especially when I press return, this causes a really noticeable lag. 

https://github.com/LazyVim/LazyVim/assets/74576123/73996620-7ab1-4fed-b0a5-5a090f938eec

I'm pretty new to both Neovim and LazyVim and was wondering if there was a way to fix this or maybe a workaround? If there's somewhere else I should be looking, that would also be helpful!

**Comment by amaanq** [comment]
probably better to ask at nvim lspconfig or the dart ls itself

**Reply by khoa-l** [reply]
I'll try that and will keep looking! As far as I can tell though there doesn't seem to be any sort of discussion forum at [dartls](https://github.com/dart-lang/sdk/tree/master/pkg/analysis_server/tool/lsp_spec) or [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/tree/master).

**Comment by BtAbdul** [comment]
can i have your config?

**Reply by Hecatoncheir** [reply]
I have same issue. In my `plugins/extras/lang/dart.lua`:

```lua
return {
  {
    "akinsho/flutter-tools.nvim",
    dependencies = { "nvim-lua/plenary.nvim" },
    config = function()
      require("flutter-tools").setup({})
    end,
  },
  {
    "nvim-neotest/neotest",
    dependencies = { "sidlatau/neotest-dart" },
    opts = function(_, opts)
      table.insert(
        opts.adapters,
        require("neotest-dart")({
          command = "fvm flutter",
          use_lsp = true,
          custom_test_method_names = { "testWidgets" },
        })
      )
    end,
  },
  {
    "mfussenegger/nvim-dap",
    config = function()
      local dap = require("dap")
      dap.adapters.dart = {
        type = "executable",
        command = "flutter",
        args = { "debug_adapter" },
      }
      dap.configurations.dart = {
        {
          type = "dart",
          request = "launch",
          name = "Launch Dart Program",
          program = "${file}",
          cwd = "${workspaceFolder}",
        },
      }
    end,
  },
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        dartls = {},
      },
    },
  },
}

```

**Comment by xiechao06** [comment]
```lua
 return {
	{
		"folke/noice.nvim",
		opts = {
			lsp = {
				progress = {
					enabled = false,
				},
			},
		},
	},
}
```
I've tried, it works
